cli/azd/internal/appdetect/docker.go

```diff
@@ -70,3 +82,50 @@ func parsePortsInLine(s string) ([]Port, error) {
 	}
 	return ports, nil
 }
+
+func addDefaultDockerfile(path string, hasParentPom bool) error {
+	if _, err := os.Stat(path); err != nil {
+		return fmt.Errorf("error accessing path %s: %w", path, err)
+	}
+
+	dockerfilePath := filepath.Join(path, "Dockerfile")
+	if _, err := os.Stat(dockerfilePath); err == nil {
+		fmt.Println("Dockerfile already exists, skipping creation.")
+		return nil
+	} else if !os.IsNotExist(err) {
+		return fmt.Errorf("error checking Dockerfile at path %s: %w", path, err)
+	}
+
+	file, err := os.Create(dockerfilePath)
+	if err != nil {
+		return fmt.Errorf("failed to create Dockerfile at %s: %w", dockerfilePath, err)
+	}
+	defer file.Close()
+
+	// for single-module project, we have to maven build first, then copy and run jar
+	// for multi-module project, just copy and run jar because of prepackage hook
+	var dockerfileContent string
+	if hasParentPom {
+		dockerfileContent = `FROM mcr.microsoft.com/openjdk/jdk:17-distroless
+COPY ./target/*.jar app.jar
+EXPOSE 8080
+ENTRYPOINT ["java", "-jar", "/app.jar"]`
+	} else {
+		dockerfileContent = `FROM maven:3 AS build
+WORKDIR /app
+COPY . .
+RUN mvn --batch-mode clean package -DskipTests
+
+FROM mcr.microsoft.com/openjdk/jdk:17-distroless
+WORKDIR /
+COPY --from=build /app/target/*.jar /app.jar
+EXPOSE 8080
```

Comment:
1. If it's a JMS /  spring cloud stream app with out spring-boot-web dependency, expose 8080 may not the right way. 
2. If in customer's application.yaml, the port is set to another one like this `server.port=8181`.
3. Now for eureka server / config server, will they work well for current port setting?

Reply:
1. I remove this line of `expose 8080` first, and then prompt user to input the desired port.
2. Currently we still not support detect the `server.port`. I think we need to enhance the port detection.

Reply:
Could you please help to create an ADO item for this?

Reply:
Yeah, I have created the item to track https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2330216, and will enhance it in another PR.

Reply:
PR merged https://github.com/azure-javaee/azure-dev/pull/94.